Install t8_cmesh_vertex_connectivity

# Bug report

**Describe the bug**
t8_cmesh_types.h includes t8-cmesh/t8-cmesh_vertex_connectivity/t8_cmesh_vertex_connectivity.h>, but the file is not installed. 

**To Reproduce**
make install and try to link with t8code. 

**Expected behavior**
Success

**Log files**
- If possible, add any output and error messages that you got
- Please upload the `config.log` file (when building with autotools)
- For failing tests, please upload the `test-suite.log` file (when building with autotools)
- When building with CMake please upload
  - `CMakeCache.txt`
  - `CMakeFiles/CMakeError.log`
  - `CMakeFiles/CMakeOutput.log`
  - `Testing/Temporary/FailedTest.log` (for failing tests)
  - `Testing/Temporary/LastTestsFailed.log` (for failing tests)
 t8code_install/include/t8_cmesh/t8_cmesh_types.h:34:10: fatal error: t8_cmesh/t8_cmesh_vertex_connectivity/t8_cmesh_vertex_connectivity.h: No such file or directory
   34 | #include <t8_cmesh/t8_cmesh_vertex_connectivity/t8_cmesh_vertex_connectivity.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

**Environment (please complete the following information):**
 - OS: [e.g. Ubuntu 22.04, Windows 11]
 - Compiler: [e.g., gcc, icc, mpicc]
 - Compiler version: [e.g. the output of 'gcc --version']
 - MPI version (if MPI is used): [The output of 'mpirun --version' or 'mpiexec --version']

**Estimated priority**
Which of these is most applicable (remove the others):

"Priority: high"  Should be solved as soon as possible 

"Priority: medium"  Should be solved within half a year 

"Priority: low" Should be solved eventually

**Additional context**
Add any other context about the problem here.

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -254,6 +254,8 @@ install( FILES
     t8_types/t8_vec.h
     t8_version.h
     t8_vtk.h
+    t8_cmesh/t8_cmesh_vertex_connectivity/t8_cmesh_vertex_connectivity.hxx
+    t8_cmesh/t8_cmesh_vertex_connectivity/t8_cmesh_vertex_connectivity.h
     TYPE INCLUDE
 )
 

